mlir/lib/Dialect/Rock/Transforms/AlignTiling.cpp

```diff
@@ -606,21 +606,24 @@ ReduceRewritePattern::matchAndRewrite(rock::ReduceOp reduceOp,
 
 void RockLinalgAlignPass::runOnOperation() {
   MLIRContext *ctx = &getContext();
+  func::FuncOp func = getOperation();
+  // Only run this pass on GPU kernel functions.
+  if (!func->hasAttr("kernel"))
+    return;
```

Comment:
Thought this check was already here.. strange

Reply:
It's there in regularize but not in align tiling